Data/models/BraveTodaySourceMO.swift

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+
+public final class BraveTodaySourceMO: NSManagedObject, CRUD {
+    @NSManaged public var enabled: Bool
+    @NSManaged public var publisherID: String
+    
+    public class func get(fromId id: String) -> BraveTodaySourceMO? {
+        getInternal(fromId: id)
+    }
+    
+    public class func all() -> [BraveTodaySourceMO] {
+        all() ?? []
+    }
+    
+    public class func setEnabled(forId id: String, enabled: Bool) {
+        setEnabledInternal(forId: id, enabled: enabled)
+    }
+    
+    public class func setEnabled(forIds ids: [String], enabled: Bool) {
+        DataController.perform(context: .new(inMemory: false)) { context in
+            ids.forEach {
+                setEnabledInternal(forId: $0, enabled: enabled, context: .existing(context))
+            }
+        }
+    }
+    
+    public class func resetSourceSelection() {
+        deleteAll()
+    }
+    
+    class func getInternal(fromId id: String, context: NSManagedObjectContext = DataController.viewContext) -> BraveTodaySourceMO? {
+        let predicate = NSPredicate(format: "\(#keyPath(BraveTodaySourceMO.publisherID)) == %@", id)
+        return first(where: predicate, context: context)
+    }
+    
+    class func setEnabledInternal(forId id: String, enabled: Bool, context: WriteContext = .new(inMemory: false)) {
+        DataController.perform(context: context) { context in
+            if let source = getInternal(fromId: id, context: context) {
+                source.enabled = enabled
```

Comment:
I was thinking that if we disable a source we could just delete it from the DB, but some sources are off by default so can't really do that right?

Reply:
Right, deleting essentially resets it back to "default" which is not what we want